state/modelconfig.go

```diff
@@ -21,11 +22,18 @@ var disallowedModelConfigAttrs = [...]string{
 
 // ModelConfig returns the complete config for the model
 func (m *Model) ModelConfig() (*config.Config, error) {
```

Comment:
One wonders why we have identical methods `Config` and `ModelConfig` on `state.Model`.

Reply:
Best guess: the work to "Move model config methods to model receiver."  when state.ModelConfig() was changed to model.ModelConfig(), the existing model.Config() wasn't looked?   The refactoring was related to CaaS implementation. (pr 7827)

Sounds like a tech-debt bug to me.. and maybe a TODO with the bug number, so folks know to use ModelConfig() moving forward?